### ISSUE TITLE
[Accessibility] Fix keyboard navigation in dropdown type editors in property grid

### ISSUE THREAD

**Issue by Tanya-Solyanik** [dotnet/winforms#12607]
Property grid drop down type editor does not support conventional keyboard navigation. Usually selection is committed when the ENTER key is pressed. However in this case value is committed on the Down arrow press. Then the drop down list is expanded, the screen reader user might want to navigate through all values in the drop down using the arrows, on each arrow key press the new item should be presented (and announced) in the selection text box, however, the  drop down is closed on the first arrow key. According to the accessibility SMEs, this is a major problem - the keyboard users are losing their work by committing a wrong value. 
Other drop down editors behave as expected 

Keyboard navigation had been designed incorrectly from the .NET Framework. 

More details in Bugs: https://github.com/dotnet/winforms/issues/12434, https://github.com/dotnet/winforms/issues/12421, https://github.com/dotnet/winforms/issues/12440, https://github.com/dotnet/winforms/issues/12031
